### PR TITLE
add support for OSX

### PR DESCRIPTION
**Pull request by tmpvar**
OSX doesn't have `clock_gettime`, but it does have `clock_get_time` which provides a monotonic clock including nanoseconds.

### CODE REVIEW

**Comment by chamnit** [comment]
Thanks Elijah!

**Comment by ashelly** [comment]
Thank you. OSX is the one platform I don't have easy access to for testing.

**Comment by tmpvar** [comment]
@ashelly cool, I use OSX primarily so, so let me know if you want me to test things in the future. 

One thing I noticed is that this thing eats 100% of 2 cpu cores - even when waiting for stdin.  I understand the want to have an emulation that is the same down to the "avr" cpu clock, especially if you want to keep the computer in sync with a running job.  However, because this eats so much cpu, you'd need a pretty beefy machine to also process the output and render some sort of visualization.

Would we be ok with adding another sim here that does it's job much like the 0.8 version (sampling instead of emulation)?  I want to write up a "learn (grbl) gcode" workshopper much like the ones in http://nodeschool.io/#workshoppers - where the challenge is to clear pockets or do contouring of parts.

**Comment by ashelly** [comment]
@tmpvar, I'm not happy with the CPU usage either.  I am thinking a better threading model would help - at least one thread spends most of its time busy-waiting - totally inefficient.  The current implementation was my first attempt at doing the timer simulation.   I arrived at this solution because I was trying avoid the problems that broke 0.8: It was so tied to the implementation details of the core Grbl code, that it broke whenever Sonny made minor adjustments.  I also wanted a realtime (or near-realtime) version for testing my gcode sender without needing hardware.

I would certainly be willing to implement a faster than real-time simulation mode. I'm open to any suggestions on how to extract  the step/position data from Grbl core without re-implementing major portions of its code,  the way the 0.8 version did.

File: deps/libuv [hunk 1]
```diff
@@ -0,0 +1 @@
+/Users/tmpvar/work/c/libuv
```
Comment (ashelly) [inline deps/libuv]: @tmpvar, is this needed for the OSX build?  If so, can it use an abreviation like `~/` instead of `/Users/tmpvar`?

**Comment by tmpvar** [comment]
gah, I totally botched this commit! I'm so sorry :(

**Pull request merged**